### DISCUSSION TITLE
Feel welcome to post questions or comments here

### DISCUSSION THREAD

**Discussion by philzook58** [philzook58/philzook58.github.io#9] [Announcements]
Discussion, good questions, comments, references is one of the benefits I derive from this blog

**Comment by marquitos0119** [comment]
I'm also a former physics guy turned computer programmer. I recently discovered formal verification methods (which is how I found your blog through your Z3 tutorial). In addition to CMBC, I've also stumbled upon [kcc](https://github.com/kframework/c-semantics) which I find very interesting for semantics analysis. In general, this topic of proving correctness of code has sent me through a rabbit hole of learning proofs, set theory, and proof assistant tools, and I think your blog has many helpful posts on the topic

**Reply by philzook58** [reply]
Glad to be of assistance! The world has a lot of neat stuff out there.

**Comment by brurucy** [comment]
Hey there Phillip, I would to know what you would have to say about the Datalog implementations on: https://github.com/brurucy/pydbsp and https://github.com/brurucy/materialized-view

Most of it stems from: https://ceur-ws.org/Vol-3801/paper1.pdf

I saw your blog post: https://www.philipzucker.com/unify/

And I think that you might find the way we implement incremental (both for additions and deletions) unification with structural sharing interesting.
